semanticdb/metacp/src/main/scala/scala/meta/internal/scalacp/TypeOps.scala

```diff
@@ -4,53 +4,41 @@ import scala.meta.internal.{semanticdb3 => s}
 import scala.meta.internal.semanticdb3.Scala._
 import scala.meta.internal.semanticdb3.SingletonType.{Tag => st}
 import scala.meta.internal.semanticdb3.SymbolInformation.{Kind => k}
-import scala.meta.internal.semanticdb3.Type.{Tag => t}
 import scala.tools.scalap.scalax.rules.scalasig._
 
 trait TypeOps { self: Scalacp =>
   implicit class XtensionTypeSType(tpe: Type) {
-    def toSemantic(linkMode: LinkMode): Option[s.Type] = {
-      def loop(tpe: Type): Option[s.Type] = {
+    def toSemantic(linkMode: LinkMode): s.Type = {
+      def loop(tpe: Type): s.Type = {
         tpe match {
           case ByNameType(tpe) =>
-            val stag = t.BY_NAME_TYPE
             val stpe = loop(tpe)
-            Some(s.Type(tag = stag, byNameType = Some(s.ByNameType(stpe))))
+            s.ByNameType(stpe)
           case RepeatedType(tpe) =>
-            val stag = t.REPEATED_TYPE
             val stpe = loop(tpe)
-            Some(s.Type(tag = stag, repeatedType = Some(s.RepeatedType(stpe))))
+            s.RepeatedType(stpe)
           case TypeRefType(pre, sym, args) =>
-            val stag = t.TYPE_REF
-            val spre = if (tpe.hasNontrivialPrefix) loop(pre) else None
+            val spre = if (tpe.hasNontrivialPrefix) loop(pre) else s.NoType
             val ssym = sym.ssym
-            val sargs = args.flatMap(loop)
-            Some(s.Type(tag = stag, typeRef = Some(s.TypeRef(spre, ssym, sargs))))
+            val sargs = args.map(loop).filter(_.isDefined)
```

Comment:
Let's skip the `isDefined` part. I made a mistake saying `flatMap` here.

Reply:
👍 It was flatMap to satisfy the `Option[Type]` signature